train.py

```diff
@@ -735,6 +795,55 @@ def train_one_epoch(
         if model_ema is not None:
             model_ema.update(model)
 
+        if loss_ is not None:
+            loss_.copy_(loss)
+            return None
+        else:
+            return loss
+
+    for batch_idx, (input, target) in enumerate(loader):
+        last_batch = batch_idx == last_idx
+        data_time_m.update(time.time() - end)
+        if not args.prefetcher:
+            input, target = input.cuda(), target.cuda()
+            if mixup_fn is not None:
+                input, target = mixup_fn(input, target)
+
+        if cg_stage is None:    # The default non-CUDAGraph mode
+            loss = _step(input, target)
+        elif cg_stage == 'get_static_shapes':
+            s = torch.cuda.Stream()
+            s.wait_stream(torch.cuda.current_stream())
+            with torch.cuda.stream(s):
+                optimizer.zero_grad(set_to_none=True)
+                loss = _step(input, target)
+            torch.cuda.current_stream().wait_stream(s)
+            return (input, target, loss)
+        elif cg_stage == 'capture':
+            s = torch.cuda.Stream()
+            s.wait_stream(torch.cuda.current_stream())
+            with torch.cuda.stream(s):
+                for _ in range(11):
```

Comment:
Any particular reason for `11`? In benchmarking you're doing `3` warmups.

Reply:
@mcarilli mentioned that cuda graph warmup with DDP needs 11 iterations instead of 3 in the cuda graph doc. https://pytorch.org/docs/stable/notes/cuda.html#id5 🤔 

The benchmark script is for single thread without DDP but training script may have DDP enabled.

Reply:
Hah, okay no problem. If Carilli says so, Carilli says so ;-)